packages/migration/src/migrations/v0043_removeTagNames.ts

```diff
@@ -14,26 +14,45 @@
  * limitations under the License.
  */
 
-function removeFontProperties({ pages, ...rest }) {
+/**
+ * External dependencies
+ */
+import type { Story, Page, TagName } from '@googleforcreators/types';
+
+/**
+ * Internal dependencies
+ */
+import type { StoryElement } from '../types';
+
+export type ElementV42 = StoryElement & {
```

Comment:
This would specifically extend TextElement rather than element, and you should use the custom type guard to filter the element list down to only text elements first.

I can see the issues though with this approach, because then you're creating a `TextElementV42` because that element type changed in the version, but what about the other element types, that don't change. It's a lot of work to find all the proper types to extend then.

Reply:
Hm, so you're saying that it should be something like:
```
const ElementV42 = 
  | GifElement
  | ImageElement
  | ProductElement
  | ShapeElement
  | TextElementV42
  | VideoElement;
```

where only the text element is extended.
And then when some other type of Element is changed in another migration file, we should again extend only that element, e.g. 
```
const ElementV39 = 
| GifElement
  | ImageElementV39
  | ProductElement
  | ShapeElement
  | TextElementV42
  | VideoElement;
```

where the image element is the new type and all the others should use the closest previously extended type (e.g. like `TextElementV42` which was extended for V42?

Do you have a simpler alternative in mind? 🤔

Reply:
On the other hand, the above is saying that all elements could have had a `tagName` property in V42, which is definitely not true. Only text elements could.

Reply:
Yep, but this would say that only text elements could, no?

```
interface TextElementV42 extends TextElement {
  tagName?: string;
}

const ElementV42 = 
  | GifElement
  | ImageElement
  | ProductElement
  | ShapeElement
  | TextElementV42
  | VideoElement;
```

Let me know if you have a simpler alternative.

Reply:
@barklund I updated the PR again to show what I meant by the code above. So basically depending on which element type is being modified, we extend only that one and leave the others as they were. It shouldn't be much more complicated than the previous version. Thoughts?